### PR TITLE
Package session start event's application information.

### PR DESCRIPTION
**Pull request by visumickey**
#no-changelog

### CODE REVIEW

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt [hunk 1]
```diff
@@ -33,14 +33,18 @@ internal constructor(
 ) {
   private val sessionGenerator = SessionGenerator(collectEvents = true)
   private val sessionCoordinator = SessionCoordinator(firebaseInstallations, backgroundDispatcher)
+  private val firebaseApp = firebaseApp
```
Comment (mrober) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt:36]: Kotlin has a shortcut for this, you can just add `val` to the firebaseApp in the primary constructor.

File: firebase-sessions/src/test/kotlin/com/google/firebase/sessions/ApplicationInfoTest.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.google.firebase.FirebaseApp
+import com.google.firebase.FirebaseOptions
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.ktx.initialize
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class ApplicationInfoTest {
+  private val MOCK_PROJECT_ID = "project"
```
[the remark this answers is not in the document]
Comment (mrober) [reply firebase-sessions/src/test/kotlin/com/google/firebase/sessions/ApplicationInfoTest.kt:15]: Also strings can have the `const` modifier.

[the remark this answers is not in the document]
Comment (visumickey) [reply firebase-sessions/src/test/kotlin/com/google/firebase/sessions/ApplicationInfoTest.kt]: Done.

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt (hunk 1 above)
Comment (visumickey) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt]: Nice comfort with Kotlin. Done.

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt [hunk 3]
```diff
@@ -27,7 +27,7 @@ import kotlinx.coroutines.CoroutineDispatcher
 
 class FirebaseSessions
 internal constructor(
-  firebaseApp: FirebaseApp,
+  val firebaseApp: FirebaseApp,
```
Comment (samedson) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt:30]: I'm curious why this change was necessary

Comment (samedson) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt:30]: Ah I see it below

**Review by samedson — APPROVED** [review]
Looks good!

File: firebase-sessions/src/test/kotlin/com/google/firebase/sessions/testing/FakeFirebaseApp.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.core.content.pm.PackageInfoBuilder
+import com.google.firebase.FirebaseApp
+import com.google.firebase.FirebaseOptions
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.ktx.initialize
+import org.robolectric.Shadows
+
+class FakeFirebaseApp {
```
Comment (mrober) [inline firebase-sessions/src/test/kotlin/com/google/firebase/sessions/testing/FakeFirebaseApp.kt:28]: Can this just be an object at the top level, instead of a class and companion object inside?

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvent.kt [hunk 5]
```diff
@@ -31,6 +31,7 @@ internal data class SessionEvent(
 
   /** Information about the session triggering the event. */
   val sessionData: SessionInfo,
+  val applicationInfo: ApplicationInfo,
```
Comment (mrober) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvent.kt:34]: Can this line have doc too? From iOS "Information about the running application."

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt (hunk 3 above)
Comment (mrober) [inline firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt:30]: Does this need to be `private val` ? Default is public in Kotlin. This class is public, but ctor is internal. I am not sure if Java will see getFirebaseApp() as a public method?

**Review by mrober — COMMENTED** [review]
LGTM just a couple comments

**Review by mrober — APPROVED** [review]
LGTM just a couple comments

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt (hunk 3 above)
Comment (visumickey) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/FirebaseSessions.kt]: Very good point and thanks for surfacing. You are right keeping it private is totally a good idea.

File: firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvent.kt (hunk 5 above)
Comment (visumickey) [reply firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionEvent.kt:36]: Done.

File: firebase-sessions/src/test/kotlin/com/google/firebase/sessions/ApplicationInfoTest.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class ApplicationInfoTest {
+
+  @Test
+  fun applicationInfo_populatesInfoCorrectly() {
+    val applicationInfo = getApplicationInfo(FakeFirebaseApp.fakeFirebaseApp())
+    Truth.assertThat(applicationInfo)
```
[the remark this answers is not in the document]
Comment (visumickey) [reply firebase-sessions/src/test/kotlin/com/google/firebase/sessions/ApplicationInfoTest.kt]: Done.

File: firebase-sessions/src/test/kotlin/com/google/firebase/sessions/testing/FakeFirebaseApp.kt (hunk 4 above)
Comment (visumickey) [reply firebase-sessions/src/test/kotlin/com/google/firebase/sessions/testing/FakeFirebaseApp.kt]: That actually is much simpler than I thought. Made the change and it looks good.
